src/env.h

```diff
@@ -1002,6 +1007,15 @@ class Environment : public MemoryRetainer {
   inline const StartExecutionCallback& embedder_entry_point() const;
   inline void set_embedder_entry_point(StartExecutionCallback&& fn);
 
+  // Set a preload function that will run before executing the entry point, this
+  // is usually used by embedders to inject scripts.
+  // The function is executed with preload(process, require), and the passed
+  // |require| function has access to internal Node.js modules.
+  // The preload function is inherited by worker threads and thus will run in
+  // work threads, so make sure the function is thread-safe.
+  inline const EmbedderPreloadCallback& embedder_preload() const;
```

Comment:
Did you intend to expose it to `node.h`? It would be more appropriate to add this to `LoadEnvironment()` and write a cctest for it, otherwise we won't have coverage for this in our tests and could break it any time.

Reply:
I added a `preload` parameter to the `node::CreateEnvironment` API and added a cctest for it.

Making it a parameter of `LoadEnvironment` would imply the preload function would only run for once, being a `CreateEnvironment` parameter implies it is part of the environment and the behavior of workers inheriting the preload function should be more intuitive.

Reply:
Conceptually `CreateEnvironment()` is meant to create a snapshotable Environment while `LoadEnvironment` would finish the initialization using run time information. If a snapshot is taken with an `Environment` that contains this callback, this needs extra code to ensure that the callback gets stored/matched in the snapshot blob somehow for `LoadEnvironment()` to run it. If the preload function needs to be run at pre-execution (I assume because it needs run time state information e.g. CLI args and environment variables) it would be simpler to make it an argument for `LoadEnvironment()`.

I am not sure about what the worker bit meant, every worker has their own Environment and they don't inherit from the main instance.

Reply:
Thanks for the elaboration, making it a `LoadEnvironment()` arg does sound more reasonable! I have updated this PR.